### PR TITLE
List security profiles applicable on a given target

### PR DESCRIPTION
**Pull request by jan-cerny**
This commit introduce 'oscapd-evaluate target-profiles' subcommand.
Similarly to 'oscapd-evaluate target-cpes', this will show a list
of available profiles to a scan target.

### CODE REVIEW

File: openscap_daemon/cli_helpers.py [hunk 1]
```diff
@@ -302,3 +302,27 @@ def summarize_standard_compliance_results(arf_source, result_list):
         result_json["Custom"] = {"XCCDF result": result}
 
         result_list.append(result_json)
+
+def get_list_of_profiles(sds_filepath):
```
Comment (mpreisler) [inline openscap_daemon/cli_helpers.py]: There already is a function for this. Please use that instead. It's called scrape_profiles or something like that.

Comment (jan-cerny) [inline openscap_daemon/cli_helpers.py]: @mpreisler  Aha OK, I was looking for something like that before I implemented this, but I must be blind. I will try to reuse that here.

**Comment by jan-cerny** [comment]
I used `oscap_helpers.get_profile_choices_for_input()` instead.

File: openscap_daemon/oscap_helpers.py [hunk 2]
```diff
@@ -125,7 +125,7 @@ def scrape_profiles(tree, namespace, dest):
             tailoring_tree, "http://checklists.nist.gov/xccdf/1.2", ret
         )
 
-    ret[""] = "(default)"
+    ret["(default)"] = "Default profile"
```
Comment (mpreisler) [inline openscap_daemon/oscap_helpers.py]: ??? This will break existing use-cases of this API.

Comment (jan-cerny) [inline openscap_daemon/oscap_helpers.py]: Thanks for suggestion. This change isn't necessary. I'll remove it.

**Comment by jan-cerny** [comment]
I have removed the commit with change of default profile. I have also fixed the std output for `oscapd-evaluate target-profiles`. I have rebased the pull request.

File: bin/oscapd-evaluate [hunk 3]
```diff
@@ -517,6 +529,20 @@ def main():
...
         sys.exit(0)
 
+    elif args.action == "target-profiles":
+        cpes = evaluation_spec.EvaluationSpec.detect_CPEs_of_target(
+            args.target, config
+        )
+        ssg_sds = config.get_ssg_sds(cpes)
+        print("Security profiles applicable on target " + args.target + ":")
+        profiles = oscap_helpers.get_profile_choices_for_input(ssg_sds, None)
+        for profile_id, title in profiles.items():
```
Comment (mpreisler) [inline bin/oscapd-evaluate]: I wonder what happens if the SDS has multiple benchmarks.

**Comment by jan-cerny** [comment]
@mpreisler That's a good question.

All profiles from all benchmarks will be printed.

The problem is that the XCCDF ID is not printed here. Moreover, `oscapd evaluate scan` has no way to choose benchmark within a datastream. Default benchmark is always chosen. 

Scan won't work for `xccdf_org.ssgproject.content_profile_pci-dss_centric` on RHEL7, even though this will be listed here.

I have encountered same behaviour of `oscapd-cli task-create`. It also allows me to create a task with profile `xccdf_org.ssgproject.content_profile_pci-dss_centric` on RHEL7 , but running the task errors out, because it can't find the profile in the datatstram.

**Comment by jan-cerny** [comment]
@mpreisler I have further investigated the issue with multiple benchmarks in one SDS. I think about 2 solutions:

1) Use only the default benchmark everywhere, don't list profiles from other benchmarks. That's would be very easy to do. I don't like the solution, because we couldn't use the PCI-DSS centric profile from SSG or something else in future.

2) Support multiple benchmarks in one datastream. That would require to rewrite some of the existing methods, namely `get_profile_choices_for_input`, so that it doesn't return a dictionary [id:title], but rather a dictionary of dicitionaries. Also implement `--bechmark-id` in `oscapd-evaluate scan`. Also implement support for multiple benchmarks in interactive task creating.

What do you think about this?

**Comment by jan-cerny** [comment]
Solution 1 is a 6-line patch. I can push the patch here as a temporary solution, and work on solution 2 in a separate pull request. The patch won't introduce a regression, because multiple benchmarks in a single datastream aren't supported now, as I mentioned before.

**Comment by mpreisler** [comment]
@jan-cerny agreed

**Comment by jan-cerny** [comment]
@mpreisler OK, I have pushed the temporary patch.

File: bin/oscapd-evaluate [hunk 4]
```diff
@@ -450,6 +450,18 @@ def main():
         "'docker-container://CONTAINER_ID', 'vm-domain://VM_NAME', "
         "'vm-image:///path/to/image.qcow2', 'chroot:///path/to/chroot'."
     )
+    target_profiles_parser = subparsers.add_parser(
+        "target-profiles",
+        help="Detect security profiles applicable on given target"
```
Comment (mpreisler) [inline bin/oscapd-evaluate]: This is specifically for SCAP Security Guide and SSG only, right? Shouldn't the help text say that?

Comment (jan-cerny) [inline bin/oscapd-evaluate]: Yes, it is, it lists profiles for SSG. Thanks for suggestion. I will improve the help.

**Comment by mpreisler** [comment]
ACK

**Pull request merged**